src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        values = (V[]) new Object[MAX_ITEMS_NUMBER];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = indexOf(key);
+        if (index == -1) {
+            if (size + 1 > MAX_ITEMS_NUMBER) {
+                throw new RuntimeException("Cannot add more elements!"
+                        + " The size of storage is " + MAX_ITEMS_NUMBER);
+            }
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            values[index] = value;
+        }
```

Comment:
1) In general when code has less tabulations it looks simpler and cleaner.
Below same code but a bit reordered
2) Please avoid +1 or -1 in conditions when case may be handled with <= or >=
```suggestion
        int index = indexOf(key);
        if (index != -1) {
            values[index] = value;
            return;
        }
        if (size >= MAX_ITEMS_NUMBER) {
            throw new RuntimeException("Cannot add more elements!"
                    + " The size of storage is " + MAX_ITEMS_NUMBER);
        }
        keys[size] = key;
        values[size] = value;
        size++;
```